content/docs/components-and-props.md

```diff
@@ -231,32 +231,34 @@ function Comment(props) {
 
-Extracting components might seem like grunt work at first, but having a palette of reusable components pays off in larger apps. A good rule of thumb is that if a part of your UI is used several times (`Button`, `Panel`, `Avatar`), or is complex enough on its own (`App`, `FeedStory`, `Comment`), it is a good candidate to be a reusable component.
+Սկզբում, կոմպոնենտների տարանջատումը, կարող է թվալ «անշնորհակալ» գործ, բայց բազմակի օգտագերծվող կոմպոնենտների պալիտրա ունենալը կարդարացնի իրեն մեծ հավելվածներում։ Հիմնական սկզբունքն այն է, որ եթե ձեր UI-ի մի մասը օգտագործվում է բազմակի անգամներ (`Button`, `Panel`, `Avatar`) կամ բավականին բարդ է ինքն իրենով (`App`, `FeedStory`, `Comment`), ապա այն լավ թեկնածու է, որպեսի դառնա բազմակի օգտագործման կոմպոնենտ։
 
-## Props are Read-Only {#props-are-read-only}
+## Prop-երը կարելի է միայն կարդալ (read-only) {#props-are-read-only}
 
-Whether you declare a component [as a function or a class](#function-and-class-components), it must never modify its own props. Consider this `sum` function:
+Անկախ ինչպես կհայտարարեք կոմպոնենտը, [որպես ֆունկցիա թե կլաս](#function-and-class-components), այն երբեք չպետք է փոփոխի իր prop-երը։
+
+Դիտարկենք այս `sum` ֆունկցիան.
 
 ```js
 function sum(a, b) {
   return a + b;
 }
 ```
 
-Such functions are called ["pure"](https://en.wikipedia.org/wiki/Pure_function) because they do not attempt to change their inputs, and always return the same result for the same inputs.
+Այսպիսի ֆունկցիաները կոչվում են [«մաքուր»](https://en.wikipedia.org/wiki/Pure_function)<sub>`eng`</sub>, որովհետև նրանք չեն փորձում փոխել իրենց մուտքային արժեքները և նույն մուտքային արժեքների դեպքում միշտ վերադարևնում են նույն արդյունքը։
 
-In contrast, this function is impure because it changes its own input:
+Ի հակադրություն դրան, այս ֆունկցիան «մաքուր» չէ, որովհետև այն փոխում է իր մուտքային արժեքները.
 
 ```js
 function withdraw(account, amount) {
   account.total -= amount;
 }
 ```
 
-React is pretty flexible but it has a single strict rule:
+React-ը բավականին ճկուն է, բայց այն ունի մեկ խիստ կանոն.
 
-**All React components must act like pure functions with respect to their props.**
+**Բոլոր React կոմպոնենտները իրենց prop-երի նկատմամբ պետք է իրենց դրսևորեն, որպես մաքուր ֆունկցիաներ**
 
-Of course, application UIs are dynamic and change over time. In the [next section](/docs/state-and-lifecycle.html), we will introduce a new concept of "state". State allows React components to change their output over time in response to user actions, network responses, and anything else, without violating this rule.
+Իհարկե, հավելվածի UI-ը դինամիկ է և փոխվում է ժամանակի ընթացքում։ [Հաջորդ բաժնում](/docs/state-and-lifecycle.html) մենք կներկայացնենք նոր կոնցեպտ - «վիճակ»(state). State-ը թույլ է տալիս React կոմպոնենտին փոխել իր ելքային արժեքները ժամանակի ընթացքում ի պատասխան օգտագործողի գործողություններին, ցանցային պատասխաններին և այլն\` չխախտելով այս կանոնը։
```

Comment:
"ի պատասխանից" առաջ ","

Reply:
միգուցե շատ անէական ա, բայց "փոխվում է ժամանակի ընթացքում" ֊> "ժամանակի ընթացքում փոփոխվում է"։ Ուղղակի էս տարբերակը  ականջիս շատ ա նստած։

Reply:
new concept of "state"֊ը կարա 2 տարբեր իմաստներ ունենա։ Զուտ ճշտեմ, որ ճիշտն ա գրված։
1․ նոր կոնցեպտ - «վիճակ»(state)
2. "վիճակ»֊ի(state) նոր կոնցեպտ/գաղափար

Reply:
այլն\` ֊> այլն`

Reply:
@SonaHarutyunyan 
1․ նոր կոնցեպտ - «վիճակ»(state) <- էս տարբերակը ճիշտ ա

\ նշանը հաջորդ սիմվոլի էկրանավորման  համար ա դրած։

Reply:
եթե "ժամանակի ընթացքում"֊ը կա, ավելի սիրուն ա "փոՓՈխել իր ելքային արժեքները ժամանակի ընթացքում"